### PR TITLE
Don't use dns by default for placement service in k8s. Get placement addresses from env variable.

### PR DESCRIPTION
**Pull request by elena-kolevska**
# Description

This PR changes the default placement service address in k8s deployments from the headless service to the addresses of the pods.
Placement addresses can also be specified through an env variable.

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [x] Code compiles correctly
* [x] Created/updated tests
* [x] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: charts/dapr/charts/dapr_placement/templates/dapr_placement_statefulset.yaml [hunk 1]
```diff
@@ -97,7 +97,7 @@ spec:
   {{- end }}
 {{- end }}
         ports:
-          - containerPort: {{ .Values.ports.apiPort }}
```
Comment (artursouza) [inline charts/dapr/charts/dapr_placement/templates/dapr_placement_statefulset.yaml:100]: Why is this needed?

**Review by artursouza — COMMENTED** [review]
minor nit

File: charts/dapr/charts/dapr_placement/templates/dapr_placement_statefulset.yaml (hunk 1 above)
Comment (elena-kolevska) [reply charts/dapr/charts/dapr_placement/templates/dapr_placement_statefulset.yaml:100]: This was a variable under the placement charts which defines the port of the placement service.
In order to be able to access it from the injector chart, I moved it to global.

**Comment by JoshVanL** [comment]
I don't fully understand the need for this PR- the benefit of using the headless service is that endpoints are added/removed from the DNS response according to each Pods health. Sure this is behaviour we want to keep?

**Comment by elena-kolevska** [comment]
Closing because the problem that this PR was trying to address was solved in https://github.com/dapr/dapr/pull/7919
